src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1070,6 +1083,15 @@ private void PrepareServices()
                 _appResources.Add(new ServiceAppResource(sp.ModelResource, svc, endpoint));
             }
         }
+
+        if (endpointsWithHostUnset.Any())
+        {
+            _logger.LogWarning("You have unproxied container endpoints without an explicit host port set. By default these endpoints will attempt to bind a host port that matches the container target port. This can lead to port conflicts if multiple containers are using the same target port(s). For containers running with a persistent lifetime or container endpoints with IsProxied disabled, we recommend specifying explicit host ports. For more information on container networking in .NET Aspire see: https://aka.ms/dotnet/aspire/container-networking");
+            foreach (var (resource, endpoint) in endpointsWithHostUnset)
+            {
+                _logger.LogInformation("'{EndpointName}' endpoint for '{ResourceName}' doesn't have a host port set. Attempting to bind host port '{TargetPort}'.", endpoint.Name, resource.Name, endpoint.TargetPort);
+            }
```

Comment:
Maybe bump these down to `LogInformation` and `LogDebug` respectively? My only concern is that `LogInformation` from `ApplicationExecutor` doesn't normally show up, so we'd only surface the guidance if a user increases their logging verbosity.

Reply:
Logs currently look like this with a warning level. DcpExecutor logs below warning don't currently get surfaced with default log levels:
![image](https://github.com/user-attachments/assets/011ffdfa-3172-49d2-b505-bc06937da9d4)

Reply:
Made a large refactor to allow default unproxied endpoints for persistent containers and support dynamic port allocation for unproxied container endpoints.

Reply:
Ended up reworking things to no longer require logging